### ISSUE TITLE
[Feature] Villager Trade

### ISSUE THREAD

**Issue by drakray** [TheExceptionist/Coherent-Villages#6]
I think having special trade for your villager could be nice, something along the line of Villager Buying Thing, like Armor/Weapon for Guard, Arrow for Archer, Lapis/Potion/Ingredient for Mage.
A way to get Emerald from them, and maybe later in the trading array, Selling Thing, like mob drop for cheap.
That would make them a sort sort "Immersive Grinding Tower"

**Comment by TheExceptionist** [comment]
Alright,
But that'll probably be a bit tricky to pull off.
So it'll have to come later

**Comment by TheExceptionist** [comment]
Doesn't make sense for a guard to sell stuff
Might allow mages and alchemist to however
Merchants already do

**Comment by drakray** [comment]
I meant "selling" emerald for stuff, as in, the Guard buy armor from the player

**Comment by TheExceptionist** [comment]
ahh, that makes more sense
Hmm... I seriously need to figure out how to fix the knight types' AI before I do anything else.

**Comment by SaucissePuree** [comment]
Yeah, or you can create the GOLD currency (realy basic, only gold) that drop from monsters, and add a vendor in the village (he would sell all kind of things)
